tools/shared.py

```diff
@@ -610,6 +614,7 @@ def check_sanity(force=False):
     # some warning, mostly not fatal checks - do them even if EM_IGNORE_SANITY is on
     check_llvm_version()
     check_node_version()
+    check_python_version()
```

Comment:
actually `check_sanity` is not the best place for this - it doesn't run every time you call `emcc`. So if you call it once with python2, then with python3, it won't check.

I think we can just call `check_python_version` right after its definition (or a better place in the top scope of `shared.py` if there is one), so it always runs once when `shared.py` is loaded. I think that should appear once per emcc invocation?

Reply:
>I think that should appear once per emcc invocation?

I'm afraid that several tests expecting empty `stderr` output will break...

Reply:
My idea is that we only emit the per-invocation warning when user directly called `emcc` instead of` python3 emcc.py`, and we keep `check_python_version` for the sanity check. AFAIK every test calls `PYTHON EMCC` rather than directly calling `emcc` so it shouldn't require complex changes. Thoughts?

```
>emcc hello.c
WARNING:root:Running on Python 3 in the experimental way
>python3 ./emcc.py hello.c
>emcc -v
WARNING:root:Configured to run on Python 3 which is experimental
```